### ISSUE TITLE
Find Wikipedia pages from Wikidata

### ISSUE THREAD

**Issue by jluttine** [osmandapp/Osmand#9423]
In short: Osmand doesn't find Wikipedia pages from Wikidata. Osmand requires Wikipedia pages to be given explicitly.

An example node with `wikidata` tag but no `wikipedia` tag, but there still exists a Wikipedia page for that Wikidata object: https://www.openstreetmap.org/way/286312540

The node subject has a wikipedia page, but that page is not defined as `wikipedia` tag for the node because the node already has `wikidata` tag defined. I would have expected Osmand to show a link to that Wikipedia page similarly as it would when `wikipedia` tag was explicitly given. That is, Osmand could find those Wikipedia pages from Wikidata when `wikidata` tag is given.

Another minor comment, not sure if I should open an issue for this: When the node is selected and I click "Details", I can see the Wikidata entry. It also seems like a link but if I click it, it actually just opens my phone app and tries to call to that number.. I would have expected a link to Wikidata page, if any link at all.

**Comment by vshcherb** [comment]
Interesting note, though OsmAnd should be able to display & search that point with POI Wikipedia layer.
The minor issue obviously should open web-browser cause we don't plan yet to support wikidata as data layer

**Comment by jluttine** [comment]
Thanks for the reply! About the POI Wikipedia layer:

1. It won't show that point if there's no location defined in Wikidata (https://www.openstreetmap.org/node/1551784121).

2. If relying on POI Wikipedia layer, Wikipedia pages wouldn't be shown as part of the object Details page, but as a separate POI. But if there is `wikipedia` tag, then it's shown in the Details page, so I'd find this feature request just an improvement of that existing feature.

I would even argue that people shouldn't tag `wikipedia` but the more general `wikidata` which will provide the Wikipedia pages in all languages that they exist. So, `wikidata` is a much better way of tagging the same information and more (e.g., it connects to many other services than just Wikipedia and provides a lot of structured metadata). That's why I think it'd be awesome if Osmand could make use of it by finding the Wikipedia pages automatically because all the needed information is already given.

**Comment by vshcherb** [comment]
We will need to store inside Wikipedia points (separate maps) tag wikidata and after that we will be able to suggest from points on the map that here is Wikipedia point.
As of today there is an automatical section that searches all wikipedia articles around that point.

**Comment by jluttine** [comment]
Just to make sure my feature request is correctly understood, I thought I'd clarify a bit: I'm requesting that the Details section of a selected POI would show Wikipedia entry even when `wikipedia` tag isn't set if `wikidata` tag is set and there are Wikipedia pages in Wikidata. So, my suggestion isn't related to the Wikipedia layer.

That is, similarly as this suburb shows Wikipedia page because it has `wikipedia` tag set but so that `wikidata` alone would be sufficient to make this happen:

![photo_2020-07-11_10-51-41](https://user-images.githubusercontent.com/2195834/87219568-061e8c80-c365-11ea-91e3-0a0166b9078e.jpg)

**Comment by Sequynth** [comment]
I agree with jluttine. Using wikidata from a POIs tags could immediatly provide the user with useful links to a specific POI. In the case of public artwork it could provide links to the artist (by looking artist:wikidata=...), or the in case of statutes the article to the depicted person (subject:wikidata=...). Using wikidata would also provide links to all language versions of an article, allowing OsmAnd to present the user with the correct language (assuming same as OsmAnd or system-language) without it being needed explicitly as an OSM tag.

**Comment by aceman444** [comment]
Yes, this would be useful. Some people only add wikidata tags to objects and it should be possible for OsmAnd to automatically retrieve the right wikipedia article from this wikidata ID.

**Comment by letypequividelespoubelles** [comment]
I was searching for this feature ! Is, while generating the map, creating a tag wikipedia=* when there is a wikidata=* tag and no wikipedia tag would be enough, as it has been made in #10462 ? I guess yes, and it shouldn't be too hard.

**Comment by Crashillo** [comment]
Wouldn't be possible to make an API call from the wikidata value? For instance, wikidata API exposes the following endpoint:

**https://www.wikidata.org/w/api.php?action=wbgetentities&format=json&props=sitelinks&ids=Q934**

That returns all wikipedia articles for the Q934 object, using `site` and `title` properties is quite easy create the links, something like:
```js
Object.entries(response.entities["Q934"].sitelinks)
  .map(([wiki, {title}]) => `https://${wiki.slice(0, -4)}.wikipedia.org/wiki/${encodeURIComponent(title)}`) 

// output
// [ "https://ace.wikipedia.org/wiki/Kut%C3%B4b%20Bar%C3%B4h", "https://af.wikipedia.org/wiki/Noordpool", "https://als.wikipedia.org/wiki/Nordpol", ...
```
